RPerformanceTracking/Private/_RPTMainThreadWatcher.m

```diff
@@ -0,0 +1,60 @@
+#import "_RPTMainThreadWatcher.h"
+#import "_RPTHelpers.h"
+#import "_RPTTrackingManager.h"
+#import "_RPTTracker.h"
+
+@interface _RPTMainThreadWatcher ()
+@property (nonatomic) BOOL                 watcherRunning;
+@property (nonatomic) dispatch_semaphore_t semaphore;
+@property (nonatomic) NSTimeInterval       blockThreshold;
+@property (nonatomic) NSTimeInterval       startTime;
+@property (nonatomic) NSTimeInterval       endTime;
+@property (nonatomic) uint_fast64_t        trackingIdentifier;
+@end
+
+// Inspired by https://medium.com/@mandrigin/ios-app-performance-instruments-beyond-48fe7b7cdf2
+@implementation _RPTMainThreadWatcher
+
+- (instancetype)init
+{
+    if (self = [super init])
+    {
+        _watcherRunning = NO;
+        _blockThreshold = 0.4;
```

Comment:
@donnie-jp 
Why Threshold is only 0.4?

Reply:
https://medium.com/@mandrigin/ios-app-performance-instruments-beyond-48fe7b7cdf2